67_SubAllocatedDescriptorSet/main.cpp

```diff
@@ -0,0 +1,290 @@
+				poolParams.maxSets = 1;
+				poolParams.flags = core::bitflag(video::IDescriptorPool::E_CREATE_FLAGS::ECF_UPDATE_AFTER_BIND_BIT);
+			}
+
+			auto descriptorPool = m_device->createDescriptorPool(std::move(poolParams));
+			auto descriptorSet = descriptorPool->createDescriptorSet(core::smart_refctd_ptr(descriptorSetLayout));
+
+
+			// TODO: I don't think these are needed for sub allocated descriptor sets (alignment isn't needed, and min size is 1)
+			auto subAllocatedDescriptorSet = core::make_smart_refctd_ptr<nbl::video::SubAllocatedDescriptorSet>(core::smart_refctd_ptr(descriptorSet), core::smart_refctd_ptr(m_device)); 
+			//std::vector<uint32_t> allocation(MaxDescriptors, core::PoolAddressAllocator<uint32_t>::invalid_address);
+
+			//std::vector<uint32_t> indices;
+			//indices.reserve(MaxDescriptors);
+			//for (uint32_t i = 0; i < MaxDescriptors; i++)
+			//	indices.push_back(i);
+
+			//auto allocNum = subAllocatedDescriptorSet->multi_allocate(AllocatedBinding, allocation.size(), allocation.data());
+			//assert(allocNum == 0);
+			m_subAllocDescriptorSet = std::move(subAllocatedDescriptorSet);
+
+			//bool response = writeDescriptors(allocation.size(), indices.data(), allocation.data());
+			//if (!response) return false;
+			
+			return true;
+		}
+
+		bool keepRunning() override { return m_iteration<MaxIterations; }
+
+		void workLoopBody() override
+		{
+			IQueue* const queue = getComputeQueue();
+
+			// Similar idea to example 05 (streaming buffers)
+			// We will be allocating and freeing stuff, latched on previous frame's timeline semaphore
+			auto rng = nbl::hlsl::Xoroshiro64StarStar::construct({ m_iteration ^ 0xdeadbeefu,std::hash<string>()(_NBL_APP_NAME_) });
+			const auto elementCount = rng() % MaxAllocPerFrame;
+			m_logger->log("elementCount: %d", system::ILogger::ELL_INFO, elementCount);
+
+			std::vector<SubAllocatedDescriptorSet::value_type> values(elementCount, SubAllocatedDescriptorSet::invalid_value);
+
+			{
+				std::chrono::steady_clock::time_point waitTill(std::chrono::years(45));
+				m_subAllocDescriptorSet->multi_allocate(waitTill, AllocatedBinding, elementCount, values.data());
+
+				std::vector<SubAllocatedDescriptorSet::value_type> indices;
+				indices.reserve(elementCount);
+				for (uint32_t i = 0; i < elementCount; i++)
+					indices.push_back(i);
+			
+				bool response = writeDescriptors(elementCount, indices.data(), values.data());
+				assert(response);
+			}
+
+			uint32_t poolIx;
+			do
+			{
+				poolIx = m_poolCache->acquirePool();
+			} while (poolIx==ICommandPoolCache::invalid_index);
+
+			smart_refctd_ptr<IGPUCommandBuffer> cmdbuf;
+			{
+				m_poolCache->getPool(poolIx)->createCommandBuffers(IGPUCommandPool::BUFFER_LEVEL::PRIMARY,{&cmdbuf,1},core::smart_refctd_ptr(m_logger));
+				cmdbuf->begin(IGPUCommandBuffer::USAGE::ONE_TIME_SUBMIT_BIT);
+
+				// COMMAND RECORDING
+				// Here we would hipothetically use the descriptors created above
```

Comment:
an idea, actually bind the descriptor set, launch an empty compute pipeline

might find some bugs in `IGPUCommandBuffer` and friends